Firestore/Source/API/FSTUserDataConverter.mm

```diff
@@ -350,13 +378,14 @@ - (FSTParsedUpdateData *)parsedUpdateData:(id)input {
 
   NSDictionary *dict = input;
 
-  NSMutableArray<FSTFieldPath *> *fieldMaskPaths = [NSMutableArray array];
+  __block std::vector<FieldPath> fieldMaskPaths{};
   __block FSTObjectValue *updateData = [FSTObjectValue objectValue];
 
   FSTParseContext *context =
-      [FSTParseContext contextWithSource:FSTUserDataSourceUpdate path:[FSTFieldPath emptyPath]];
+      [FSTParseContext contextWithSource:FSTUserDataSourceUpdate
+                                    path:absl::make_unique<FieldPath>(FieldPath::EmptyPath())];
   [dict enumerateKeysAndObjectsUsingBlock:^(id key, id value, BOOL *stop) {
-    FSTFieldPath *path;
+    FieldPath path{};
```

Comment:
Throughout this PR, there's a number of places where your initializing using an empty initializer list rather than just the default ctor, i.e.:

FieldPath path{};  or vector<string> v{};
vs
FieldPath path; or vector<string> v;

Is there a reason for this? (I don't have an objection... but coming from a mostly pre-11 background, I'm curious if there's an advantage to this.)

Reply:
Just want to make explicit it is defined and initialized here.